client/state/oauth2-clients/ui/reducer.js

```diff
@@ -6,11 +6,15 @@ export const currentClientId = ( state = null, action ) => {
 	switch ( action.type ) {
 		case ROUTE_SET: {
 			const { path, query } = action;
-			if ( startsWith( path, '/log-in' ) ) {
-				return query.client_id ? Number( query.client_id ) : state;
+			if ( startsWith( path, '/log-in' ) && query.client_id ) {
+				return Number( query.client_id );
 			}
 
-			if ( startsWith( path, '/start/wpcc' ) || startsWith( path, '/start/crowdsignal' ) ) {
+			if (
+				startsWith( path, '/log-in/apple/callback' ) ||
```

Comment:
Interesting! @rjchow Do you remember if there was an existing account with the same email before you sign up with Apple?

When I tried Apple signup flow, I remember that I chose "Hide My Email" option and it did create a new account for me. 🤔 However, I just tried to disconnect Apple and re-sign up but chose "Share My Email" (I have an account with that email.), and I got the account not found error.

I'm wondering if it's because we don't automatically connect Apple ID for an existing account. AFAIK, we do post a signup request during the Apple login callback. 

https://github.com/Automattic/wp-calypso/blob/c0b28f88134d15f7c15b72fdf0006a4499ac012a/client/server/api/sign-in-with-apple.js#L39-L50

Reply:
> Even after going into your Apple "Hide my email" settings to disconnect it from the other end, it still seems to think that I have an associated apple account .... which might be considered a bug even for the wordpress.com implementation.

Hey @Automattic/ganon, we're adding SSO options for woocmmerce.com, and I just added you to PR reviewer list as you seem like the overlord of WP.com signup and onboarding. 😄 Would you be able to confirm/review this? Thank you! 🙏

Reply:
> Do you remember if there was an existing account with the same email before you sign up with Apple?

Nope, this was the first time I used sign up with Apple!

Reply:
Okay, @rjchow I found out that it did not work because the flow name is set to `no-signup`. 😓 

https://github.com/Automattic/wp-calypso/blob/c0b28f88134d15f7c15b72fdf0006a4499ac012a/client/server/api/sign-in-with-apple.js#L12

However, I think it's expected for the login flow. So for signup flow, I changed the callback redirection logic in https://github.com/Automattic/wp-calypso/pull/67537/commits/ab40e6650a75a3e0494b26240266c4bc078980dd so that it can redirect users to `start/wpcc` path, and use the same way as Google signup.

Reply:
It should work now! But I'm not sure how to test it again because I already have an account with my apple ID. 😞  

@moon0326 or @ilyasfoo Could you help test Apple signup flow? 😄